app/code/Magento/Bundle/Model/Plugin/Frontend/Product.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Bundle\Model\Plugin\Frontend;
+
+use Magento\Bundle\Model\Product\Type;
+use Magento\Catalog\Model\Product as CatalogProduct;
+
+class Product
+{
+    /**
+     * @var Type
+     */
+    private $type;
+
+    /**
+     * @param Type $type
+     */
+    public function __construct(Type $type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * Add child identities to product identities
+     *
+     * @param CatalogProduct $product
+     * @param array $identities
+     * @return string[]
+     */
+    public function afterGetIdentities(
+        CatalogProduct $product,
+        array $identities
+    ) {
+        foreach ($this->type->getChildrenIds($product->getEntityId()) as $optionId => $childIds) {
```

Comment:
Unused variable $optionId

Reply:
Thanks for your comments, I made the code more strict and removed the unused variables.